content/tutorials/hello-world-node.md

```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
 
-##### Producer Code
+### Importing Dependencies
 
-Add the following code in the _produce.js_ file:
+```TypeScript
+// Import Fluvio class and library resources;
+import Fluvio, { FluvioAdmin, Options, TopicProducer, PartitionConsumer } from "@fluvio/client";
+
+// Use Node.js `readline` module to read from command line;
+import { createInterface, Interface } from "readline";
 
-```js
-const FluvioClient = require('@fluvio/client');
- 
-async function produceMessage() {
-  try {
-    const flvConnection = await FluvioClient.connect();
-    let replica = await flvConnection.replica("my-topic", 0);
-    let len = await replica.produce("test");
-    
-    console.log("OK: %d bytes sent", len);
-  } catch (e) {
-    console.log("error: ", e.msg());
-  }
-}
 
-await produceMessage();
 ```
 
-In summary:
+### Create the `StreamingApp` Class
 
-* _require(@fluvio/client)_ loads library into _FluvioClient_ constant. 
-* _FluvioClient.connect()_ returns the connection to the cluster.
-  * connect reads the cluster parameters from _default profile_.
-* _flvConnection.replica(...)_ looks-up _replica_ for the topic/partition.
-* _replica.produce(...)_ send a message to the _cluster_.
+```TypeScript
 
-##### Run Producer
+// Create a StreamingApp Class to encapsulate our logic;
+export default class StreamingApp {
+  // The StreamingApp class has a `fluvio` client property
+  // to produce and consumer topics;
+  fluvio: Fluvio;
 
-Run _produce.js_ to send "test" to topic/partition _my-topic/0_ :
+  // The StreamingApp class has an `admin` FluvioAdmin
+  // to manage topics;
+  admin?: FluvioAdmin;
 
-```bash
-$ node ./src/produce.js  
-OK: 4 bytes sent
-```
+  // Use Node.js' `readline` to relay messages to the topic producer
+  // from the terminal;
+  rl: Interface;
 
-To generate additional data entries, call _node ./src/produce.js_ multiple times.
+  // Setup a topic producer;
+  producer?: TopicProducer;
 
+  // Setup a topic consumer;
+  consumer?: PartitionConsumer;
 
-#### Create Consumer
+  constructor(options: { fluvio: Options }) {
+    // Create a new Fluvio Client;
+    this.fluvio = new Fluvio(options.fluvio);
 
-Inside your _src_ directory, and add a _consume.js_ file: 
+    // Setup readline
+    this.rl = createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+  }
+}
 
+```
 
-```bash
-$ tree -L 2
-.
-├── node_modules
-│   └── @fluvio
-├── package-lock.json
-├── package.json
-└── src
-    ├── consume.js
-    └── produce.js
+### Connect to the Fluvio Cluster & Setup Topic
+
+```TypeScript
+// Create connection to cluster and setup fluvio clients;
+  private async configure(
+    topicName: string,
+    partition?: number,
+    isProducer?: boolean
+  ) {
+    console.log("Connecting to Fluvio Cluster");
+    // Setup the connection for the fluvio client;
+    await this.fluvio.connect();
+
+    if (isProducer) {
```

Comment:
I think this is too much for hello world.  It should not have unnecessary steps for checking.  Just create producer

Reply:
No admin API should be used in the hello world example

Reply:
Yeah, I think it's perfectly fine for the Hello World example to lean on the CLI for admin functionality. You can have them run commands like `fluvio topic create greetings` as part of the setup with no trouble. That will simplify the code they see.

Reply:
commit 3a13337 uses fluvio cli to create topic